pkg/isoeditor/nmstate_handler.go

```diff
@@ -0,0 +1,114 @@
+package isoeditor
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -package=isoeditor -destination=mock_nmstate_handler.go . NmstateHandler
+type NmstateHandler interface {
+	CreateNmstateRamDisk(nmstatectlPath, ramDiskPath string) error
+	ExtractNmstatectl(rootfsPath string) (string, error)
```

Comment:
The way these functions are written looks like they shouldn't be separate.

Specifically how one creates a tempdir and the other cleans it up is liable to leak resources when things go wrong.

If you want to split them out for clarity in the private implementation, that's fine, but ensure you're managing the `nmstateDir` correctly.